src/parsePref.ts

```diff
@@ -12,12 +12,9 @@ export type HostedPackageSpec = ({
     committish: string,
   },
   normalizedPref: string,
-} & ({
   gitCommittish: string | null,
-} | {
-  gitCommittish: null,
-  gitRange: string,
-}))
+  gitRange?: string,
+})
```

Comment:
i'm no typescript lover, and without these changes, I was upsetting typescript
i can change it to whatever pleases typescript and the maintainers

```
src/index.ts(32,110): error TS2339: Property 'gitRange' does not exist on type '({ fetchSpec: string; hosted?: { type: string; shortcut: string; sshUrl: string; user: string; pr...'.
  Property 'gitRange' does not exist on type '{ fetchSpec: string; hosted?: { type: string; shortcut: string; sshUrl: string; user: string; pro...'.
src/index.ts(56,37): error TS2339: Property 'gitRange' does not exist on type '({ fetchSpec: string; hosted?: { type: string; shortcut: string; sshUrl: string; user: string; pr...'.
  Property 'gitRange' does not exist on type '{ fetchSpec: string; hosted?: { type: string; shortcut: string; sshUrl: string; user: string; pro...'.
src/index.ts(68,92): error TS2339: Property 'gitRange' does not exist on type '({ fetchSpec: string; hosted?: { type: string; shortcut: string; sshUrl: string; user: string; pr...'.
  Property 'gitRange' does not exist on type '{ fetchSpec: string; hosted?: { type: string; shortcut: string; sshUrl: string; user: string; pro...'.
src/index.ts(71,90): error TS2339: Property 'gitRange' does not exist on type '({ fetchSpec: string; hosted?: { type: string; shortcut: string; sshUrl: string; user: string; pr...'.
  Property 'gitRange' does not exist on type '{ fetchSpec: string; hosted?: { type: string; shortcut: string; sshUrl: string; user: string; pro...'.
```

Reply:
ok, maybe it's fine, I'll check after merging